app/modules/site_settings/models.py

```diff
@@ -466,9 +447,55 @@ def _get_default_value(cls, key):
         return def_val
 
     @classmethod
-    def set_key_value(cls, key, value):
+    def set_after_validation(cls, key, value):
+        assert key in cls.HOUSTON_SETTINGS.keys()
+        key_data = cls.HOUSTON_SETTINGS[key]
+
+        file_upload_guid = None
+        string = None
+        data = None
+        boolean = None
+        val_type = key_data['type']
+        is_public = key_data.get('public', True)
+        if val_type == dict or val_type == list:
+            data = value
+        elif val_type == bool:
+            assert isinstance(value, bool)
+            boolean = value
+        elif val_type == str:
+            assert isinstance(value, str)
```

Comment:
i think we might have the same issue here as we discussed with _boolean_.   should we be able to set the `string` field to `None`?  i guess we can go with the same idea that _"the solution is to delete it instead"_.   but i am going to ping FE team about this.

Reply:
in slack, emily responded:

> When I make changes at /admin/settings , the POST to /api/v1/site-settings/main/ includes
> ```
> email_service: null
> email_service_password: null
> email_service_username: null
> ```
> in the request payload

but that _other_ strings when cleared actually pass empty string (`""`).   not sure if this means we should let `null`/`None` be a valid value to set for strings at least.   maybe yes?

Reply:
Good point, this need more testing I think. Converting to draft to prevent merging.